nio-impl/src/main/java/org/xnio/nio/WorkerThread.java

```diff
@@ -334,7 +334,8 @@ void handleReady(final int ops) {
                     }
                 }
             } catch (IOException e) {
-                selectorLog.tracef("ConnectHandle.handleReady Exception, %s", e);
+                selectorLog.tracef("ConnectHandle.handleReady Exception, %s. Connection attempt to %s failed", e,
+                        this.connection.getPeerAddress().toString());
```

Comment:
I'd skip the `toString` part: by explicitly calling that method, you are paying the cost of stringifying the address even when trace logging is turned off.